### PR TITLE
acl:agentClass foaf:Agent and acl:AuthenticatedAgent are now respected.

### PR DESCRIPTION
**Pull request by dbernstein**
**acl:agentClass foaf:Agent and acl:AuthenticatedAgent are now respected.**
* * *

**JIRA Ticket**:  https://jira.duraspace.org/browse/FCREPO-2863

* Other Relevant Links (Mailing list discussion, related pull requests, etc.)

# What does this Pull Request do?
Ensures that acl:AuthenticatedAgent and foaf:Agent are respected.

# How should this be tested?
Includes unit as well as integration tests.
For manual testing: 
Create a foaf-agent.ttl :
```
@prefix acl: <http://www.w3.org/ns/auth/acl#> .
@prefix foaf: <http://xmlns.com/foaf/0.1/> .

<#foafAgent> a acl:Authorization ;
              acl:agentClass foaf:Agent;
              acl:mode acl:Read;
              acl:accessTo </rest/foaf-agent> .
```

Create foaf resource and acl and perform GET verifying a 200 response.
```
curl -v -u fedoraAdmin:fedoraAdmin -X PUT  http://localhost:8080/rest/foaf-agent;
curl -v -u fedoraAdmin:fedoraAdmin -X PUT --data-binary  "@foaf-agent.ttl" -H "Content-Type: text/turtle" http://localhost:8080/rest/foaf-agent/fcr:acl;
curl -v -u testuser:testpass -X GET  http://localhost:8080/rest/foaf-agent;

```
Create authenticated-agent.ttl
```
@prefix acl: <http://www.w3.org/ns/auth/acl#> .

<#authenticated> a acl:Authorization ;
              acl:agentClass acl:AuthenticatedAgent;
              acl:mode acl:Read;
              acl:accessTo </rest/authenticated-agent> .
```
Create authenticated-agent resource and  acl, perform GET and verify 200
```
curl -v -u fedoraAdmin:fedoraAdmin -X PUT  http://localhost:8080/rest/authenticated-agent
curl -v -u fedoraAdmin:fedoraAdmin -X PUT --data-binary  "@authenticated-agent.ttl" -H "Content-Type: text/turtle" http://localhost:8080/rest/authenticated-agent/fcr:acl
curl -v -u testuser:testpass -X GET  http://localhost:8080/rest/authenticated-agent
```

# Additional Notes:
Once https://github.com/fcrepo/Fedora-API-Test-Suite/pull/181 is merged in, the two related tests that were failing, 5.0-D and 5.0-E, should now pass. 

* Does this change require documentation to be updated? yes
* Does this change add any new dependencies? No

# Interested parties
Tag (@ mention) interested parties or, if unsure, @fcrepo4/committers

### CODE REVIEW

File: fcrepo-auth-webac/src/test/java/org/fcrepo/integration/auth/webac/WebACRecipesIT.java [hunk 1]
```diff
@@ -1177,4 +1177,25 @@ public void testAppendOnlyToBinary() throws IOException {
         assertEquals(HttpStatus.SC_FORBIDDEN, getStatus(postReq));
     }
 
+    @Test
+    public void testFoafAgent() throws IOException {
+        final String path = ingestObj("/rest/foaf-agent");
+        ingestAcl("fedoraAdmin", "/acls/03/foaf-agent.ttl", path + "/fcr:acl");
```
Comment (awoods) [inline fcrepo-auth-webac/src/test/java/org/fcrepo/integration/auth/webac/WebACRecipesIT.java]: For consistency, I would suggesting putting these two new TTL files in `/acls/25/`.

File: fcrepo-auth-webac/src/test/java/org/fcrepo/integration/auth/webac/WebACRecipesIT.java [hunk 2]
```diff
@@ -1177,4 +1177,25 @@ public void testAppendOnlyToBinary() throws IOException {
...
+
+        final HttpGet req = new HttpGet(path);
+        setAuth(req, username);
+        assertEquals(HttpStatus.SC_OK, getStatus(req));
+    }
+
+    @Test
+    public void testAuthenticatedAgent() throws IOException {
+        final String path = ingestObj("/rest/authenticated-agent");
+        ingestAcl("fedoraAdmin", "/acls/03/authenticated-agent.ttl", path + "/fcr:acl");
```
Comment (awoods) [inline fcrepo-auth-webac/src/test/java/org/fcrepo/integration/auth/webac/WebACRecipesIT.java]: For consistency, I would suggesting putting these two new TTL files in `/acls/25/`.

File: fcrepo-auth-webac/src/test/java/org/fcrepo/integration/auth/webac/WebACRecipesIT.java [hunk 3]
```diff
@@ -1177,4 +1177,25 @@ public void testAppendOnlyToBinary() throws IOException {
...
     }
 
+    @Test
+    public void testFoafAgent() throws IOException {
+        final String path = ingestObj("/rest/foaf-agent");
+        ingestAcl("fedoraAdmin", "/acls/03/foaf-agent.ttl", path + "/fcr:acl");
+        final String username = "user1";
+
+        final HttpGet req = new HttpGet(path);
+        setAuth(req, username);
```
Comment (awoods) [inline fcrepo-auth-webac/src/test/java/org/fcrepo/integration/auth/webac/WebACRecipesIT.java]: Maybe a comment should be added to the `setAuth()` line noting that actually, no auth headers should be set for this test... that is the point of "acl:agentClass foaf:Agent".

File: fcrepo-auth-webac/src/test/java/org/fcrepo/integration/auth/webac/WebACRecipesIT.java (hunk 1 above)
Comment (dbernstein) [reply fcrepo-auth-webac/src/test/java/org/fcrepo/integration/auth/webac/WebACRecipesIT.java]: I think /acls/26 for foaf-agent.ttl and authenticated-agent.ttl makes more sense.

**Pull request merged**